docs/csharp/nullable-migration-strategies.md

```diff
@@ -86,3 +86,120 @@ This doesn't mean you can't use a nullable type (either value type or reference
 What it does mean is that you can't use `T?` in a generic class or method declaration without constraints. For example, <xref:System.Linq.Enumerable.FirstOrDefault%60%601(System.Collections.Generic.IEnumerable%7B%60%600%7D)?displayProperty=nameWithType> won't be changed to return `T?`. You can overcome this limitation by adding either the `struct` or `class` constraint. With either of those constraints, the compiler knows how to generate code for both `T` and `T?`.
 
 You may want to restrict the types used for a generic type argument to be non-nullable types. You can do that by adding the `notnull` constraint on that type argument. When that constraint is applied, the type argument must not be a nullable type.
+
+## Updating Data Transfer Objects (DTOs) to indicate nullability
```

Comment:
I'd consider making this guidance a bit more generalized - it's relevant to any late-initialization property scenario, not just to DTOs. DTO is indeed the prominent case where this occurs, but isn't the only one.

Reply:
Late-initialized properties sounds good to me. I'd definitely avoid record types here, and also constructor-less types (since there may be a constructor, just not setting a particular property).

I think it's good to mention DTOs as well (the title can be something like "Late-initialized properties, DTOs and nullability").